olive/cli/quantize.py

```
@@ -123,9 +123,14 @@ def _get_pass_list(self, precision, algo, impl, is_hf_model):
                 and (algo is None or algo in pinfo.supported_algorithms)
                 and (precision is None or precision in pinfo.supported_precisions)
                 and (not self.args.use_qdq_encoding or "qdq" in pinfo.supported_quantization_encodings)
-                and self._check_data_name_arg(pinfo)
```

I think this condition was added to make passes requiring data config to be optional. With this change, data config is not optional?

---

Or maybe the optional logic is not valid anymore since we got rid of pass flows?

---

I move this check inside this if condition in order to throw a specific data config error. What do you mean?

---

previously it did not fail if there is no data config and just chose other valid passes. At least at the beginning, @shaahji can confirm is it's true. 
Maybe the recent refactor in #1666 changed this behavior. 

if the issue is that there is no valid pass at the end, we could throw the error at the end of the loop and just debug log why the pass requiring data configs were ignored.

---

I raised similar [issue](https://github.com/microsoft/Olive/pull/1666#discussion_r1985785085) on #1666 as well.
See @devang-ml's response on the same issue.

I don't think this change modifies the behavior except for reporting an "explicit" error for cases where a pass requires a data config but none was provided. In previous implementation, it would have reported that pass list came out empty which would be less helpful to the user.

To keep the previous behavior intact, you could report this as a warning.

---

I see. I don't mean to change this behavior, my goal is to remind the user that a specific pass requires a dataset but no dataset passing to the cli, instead of throw not supporting error.

---

In summary, presence of data name influences which quantization pass is picked - static or dynamic ?